### DISCUSSION TITLE
rust-analyzer and LSPs installed with cargo don't attach

### DISCUSSION THREAD

**Discussion by nicolasdumitru** [VonHeikemen/lsp-zero.nvim#330] [Q&A] [answered]
Hello! I have a problem specifically with rust-analyzer and other LSPs installed with rust's cargo don't attach. I don't know why and I have no such trouble other LSPs. Another LSP I have installed and behaves similarly is `svls` (I installed it through the AUR - basically it was built with cargo).

Running `:LspInfo` shows the following information for rust:

> Language client log: /home/nick/.local/state/nvim/lsp.log
>  Detected filetype:   rust
>  
>  0 client(s) attached to this buffer: 
>  
>  Other clients that match the filetype: rust
>  
>  Config: rust_analyzer
>  	filetypes:         rust
>  	root directory:    Asynchronous root_dir functions are not supported in :LspInfo
>  	cmd:               /usr/bin/rust-analyzer
>  	cmd is executable: true
>  	autostart:         true
>  	custom handlers:   
>  
>  Configured servers list: clangd, svls, pyright, bashls, lua_ls, rust_analyzer
> 

Here is a link to my LSP-Zero [configuration](https://github.com/nicolasdumitru/neovim-configuration/blob/f45eca82df02828c949e31433d56f58239d1db40/lua/nick/plugin/spec/lsp-zero.lua).

How should I fix this?

**Comment by VonHeikemen** [comment, marked as answer]
These server maybe can't detect the "root directory" of your project. Search the documentation in lspconfig [server_configurations](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md) see the section about `root_dir`.

Some servers only need to detect a `.git` folder in the root of your project to work properly.

**Reply by nicolasdumitru** [reply]
It worked. Thank you!
